plugin.js

```diff
@@ -8,6 +8,20 @@ const { Signer, sign, unsign } = require('./signer')
 const kReplySetCookies = Symbol('fastify.reply.setCookies')
 const kReplySetCookiesHookRan = Symbol('fastify.reply.setCookiesHookRan')
 
+function fastifyCookieGetCookie (name) {
+  if (!this[kReplySetCookies]?.size) return null
+
+  const cookies = []
+
+  for (const c of this[kReplySetCookies].values()) {
+    if (c.name.split(';', 1)[0] === name) {
+      cookies.push(c)
+    }
+  }
```

Comment:
Do we need to split?

```suggestion
  const namePref = `${name};`
  for (const c of this[kReplySetCookies].values()) {
    if (c.name.startsWith(namePref)) {
      cookies.push(c)
    }
  }
```

Maybe a benchmark would be good. Implement an example with getCookie. And then bench via autocannon.